Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java

```diff
@@ -0,0 +1,273 @@
+            selectedEntity = externalEntities.get(this.selectedEntityName);
+        }
+    }
+    
+    public BusinessEntity getSelectedEntity(){
+        return selectedEntity;
+    }
+    
+    public String getSelectedEntityDescription(){
+        String description = "--";
+        
+        if(selectedEntity != null && selectedEntity.getDescription() != null && !selectedEntity.getDescription().isEmpty()) {
+            description = selectedEntity.getDescription().get(0).getValue();
+        }
+        
+        return description;
+    }
+    
+    public String getSelectedEntityRegions(){
+        String regions = "--";
+        if(selectedEntity != null && selectedEntity.getCategoryBag() != null
+                && selectedEntity.getCategoryBag().getKeyedReference() != null
+                && !selectedEntity.getCategoryBag().getKeyedReference().isEmpty()) {
+            StringBuilder regionBuilder = new StringBuilder();
+            for(KeyedReference ref : selectedEntity.getCategoryBag().getKeyedReference()){
+               if(ref.getTModelKey().equals(REGION_TMODEL_KEY)) {
+                   regionBuilder.append(ref.getKeyName()).append(", ");
+               }
+            }
+            if(regionBuilder.length() > 0) {
+                regions = regionBuilder.substring(0, regionBuilder.length() - 2);
+            }
+        }
+        return regions;
+    }
+    
+    public String getSelectedEntityContact(){
+        String contactValue = "--";
+        
+        if(selectedEntity != null && selectedEntity.getContacts() != null && selectedEntity.getContacts().getContact() != null
+                && !selectedEntity.getContacts().getContact().isEmpty()) {
+            Contact contact = selectedEntity.getContacts().getContact().get(0);
+            
+            if(contact.getPersonName() != null && !contact.getPersonName().isEmpty()) {
+                contactValue = contact.getPersonName().get(0).getValue();
+            }
+        }
+        
+        return contactValue;
+    }
+    
+    public String getSelectedEntityHcid(){
+        String hcid = "--";
+        if(selectedEntity != null && selectedEntity.getIdentifierBag() != null
+                && selectedEntity.getIdentifierBag().getKeyedReference() != null
+                && !selectedEntity.getIdentifierBag().getKeyedReference().isEmpty()) {
+           for(KeyedReference ref : selectedEntity.getIdentifierBag().getKeyedReference()) {
+               if(ref.getTModelKey().equals(HCID_TMODEL_KEY)) {
+                   hcid = ref.getKeyValue();
+               }
+           } 
+        }
+        return hcid;
+    }
+    
+    public void ping(){
+        if(selectedEndpoint != null) {
+            boolean status = pingService.ping(selectedEndpoint.getServiceUrl());
+            Date timestamp = new Date();
+            EndpointManagerCache.getInstance().addOrUpdateEndPoint(selectedEndpoint.getServiceUrl(), timestamp, status);
```

Comment:
Can we just pass "new Date()" directly in, instead of using an intermediate variable?